packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -869,13 +879,13 @@ describe("BnsConnection", () => {
       const nonce = await connection.getNonce({ pubkey: faucet.pubkey });
       const signed = await profile.signTransaction(mainWalletId, faucet, sendTx, bnsCodec, nonce);
       const response = await connection.postTx(bnsCodec.bytesToPost(signed));
-      const blockInfo = await response.blockInfo.waitFor(info => info.state === BcpTransactionState.InBlock);
-      const txHeight = (blockInfo as BcpBlockInfoInBlock).height;
+      const blockInfo = await response.blockInfo.waitFor(info => info.state === TransactionState.Succeeded);
```

Comment:
What about `.waitFor(transactionProcessed)` and then assert?

As written, if the transaction fails, it will hang forever.
Using my suggestion, we can check the state and provide a much clearer error message in case the tx ever fails (including the log/msg from the blockchain to help debug)

Reply:
> As written, if the transaction fails, it will hang forever.

Yes, I'm aware of this. This is fine for tests which time out after 15 seconds